clang/lib/StaticAnalyzer/Core/CallEvent.cpp

```diff
@@ -923,12 +923,31 @@ SVal AnyCXXConstructorCall::getCXXThisVal() const {
   return UnknownVal();
 }
 
+static bool isWithinStdNamespace(const Decl *D) {
```

Comment:
I think this function could be useful for other checkers as well; consider moving this to an externally visible location (e.g. turn this into a method of `Decl`).

Reply:
Makes sense. I think I'd prefer moving this utility in a followup patch, as it may take a while to reach consensus where to put it. And I don't want to block this PR until that happens.

Reply:
Of course, feel free to leave it for a followup patch.

Reply:
@AaronBallman Do you think we clang already has something like this? Or it would make sense to have this part of Decl?

Reply:
We've already got one of these at home: https://github.com/llvm/llvm-project/blob/ee57ce57d8094026e2795182758bc57027a72293/clang/include/clang/AST/DeclBase.h#L488

Reply:
AFAIK not quite. Mine works even if the class a nested subclass within the std namespace, such as "std::vector::iterator". So directly using `isInStdNamespace` wouldn't quite cut it.

Reply:
Huh, that behavior would be surprising to me because nested subclasses are not in the `std` namespace. (Not saying it's wrong for the static analyzer, but I would not expect that behavior when doing code reviews, for example.)